apps/src/templates/teacherDashboard/SectionActionDropdown.jsx

```diff
@@ -0,0 +1,271 @@
+import Button from '../Button';
+import DialogFooter from "./DialogFooter";
+
+const styles = {
+  actionButton:{
+    border: '1px solid ' + color.white,
+    paddingLeft: 5,
+    paddingRight: 5,
+    paddingTop: 4,
+    paddingBottom: 4,
+    borderRadius: 5,
+    color: color.lighter_gray,
+    margin: 3
+  },
+  hoverFocus: {
+    backgroundColor: color.lighter_gray,
+    border: '1px solid ' + color.light_gray,
+    borderRadius: 5,
+    paddingTop: 4,
+    paddingBottom: 4,
+    paddingLeft: 5,
+    paddingRight: 5,
+    color: color.white,
+  },
+  xIcon: {
+    paddingRight: 5,
+  },
+  heading: {
+    borderTopWidth: 0,
+    borderBottomWidth: 1,
+    borderRightWidth: 0,
+    borderLeftWidth: 0,
+    borderStyle: 'solid',
+    borderColor: color.default_text,
+    paddingBottom: 20,
+    marginBottom: 30,
+  },
+};
+
+class SectionActionDropdown extends Component {
+  static propTypes = {
+    handleEdit: PropTypes.func,
+    sectionData: sortableSectionShape.isRequired,
+
+    //Provided by redux
+    removeSection: PropTypes.func.isRequired,
+    toggleSectionHidden: PropTypes.func.isRequired,
+    sectionCode: PropTypes.string,
+    sectionName: PropTypes.string,
+    updateRoster: PropTypes.func.isRequired,
+  };
+
+  state = {
+    deleting: false,
+    open: false,
+    canOpen: true,
+    menuTop: 0, // location of dropdown menu
+    menuLeft: 0,
+    currWindowWidth: window.innerWidth, // used to calculate location of menu on resize
+  };
+
+  // Menu open
+  open = () => {
+    this.updateMenuLocation();
+    window.addEventListener("resize", this.updateMenuLocation);
+    this.setState({open: true, canOpen: false});
+  }
+
+  // Menu closed
+  close = () => {
+    window.removeEventListener("resize", this.updateMenuLocation);
+    this.setState({open: false});
+  }
+
+  // Menu closed
+  beforeClose = (_, resetPortalState) => {
+    resetPortalState();
+    this.setState({open: false});
+    window.setTimeout(() => this.setState({canOpen: true}), 0);
+  };
+
+  onConfirmDelete = () => {
+      const {removeSection } = this.props;
+      const section = this.props.sectionData;
+      $.ajax({
+          url: `/v2/sections/${section.id}`,
+          method: 'DELETE',
+      }).done(() => {
+          removeSection(section.id);
+      }).fail((jqXhr, status) => {
+          // We may want to handle this more cleanly in the future, but for now this
+          // matches the experience we got in angular
+          alert(i18n.unexpectedError());
+          console.error(status);
+      });
+  };
+
+  onClickEdit = () => {
+      this.props.handleEdit(this.props.sectionData.id);
+  };
+
+  onClickHideShow = () => {
+      this.props.toggleSectionHidden(this.props.sectionData.id);
+  };
+
+  onClickSync = () => {
+    // Section code is the course ID, without the G- or C- prefix.
+    const courseId = this.props.sectionCode.replace(/^[GC]-/, '');
+    this.props.updateRoster(courseId, this.props.sectionName);
```

Comment:
There seem to be a lot of callbacks in this component that aren't covered by our tests.  Some unit tests that simulate button clicks, etc, and ensure all this works as expected would be nice.

Reply:
Discussed off-line. Portal Compatibility issue with our current enzyme version.